ui/pages/confirmations/components/confirm/info/shared/redesigned-gas-fees/use-fees-calculations.ts

```diff
@@ -0,0 +1,167 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { useEffect, useMemo, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { EtherDenomination } from '../../../../../../../../shared/constants/common';
+import {
+  addHexes,
+  getEthConversionFromWeiHex,
+  getValueFromWeiHex,
+  multiplyHexes,
+} from '../../../../../../../../shared/modules/conversion.utils';
+import { getConversionRate } from '../../../../../../../ducks/metamask/metamask';
+import { useFiatFormatter } from '../../../../../../../hooks/useFiatFormatter';
+import { getCurrentCurrency } from '../../../../../../../selectors';
+
+export function useFeeCalculations(
+  hasLayer1GasFee: boolean,
+  layer1GasFee: string,
+  gasEstimate: string,
```

Comment:
For clarity, are we being strict with `string` versus `Hex` so we know which is a decimal string for example?